app/lib/src/lib.rs

```diff
@@ -0,0 +1,28 @@
+// in Wii mode, enable `no_std` and import ogc_rs replacements of common functions.
+#![cfg_attr(feature = "wii", no_std)]
+#[cfg(feature = "wii")]
+pub use ogc_rs::{print, println};
+
+#[cfg(not(feature = "wii"))]
+pub use std::{print, println};
+
+
+fn foo() -> bool {
+    println!("Hello, world!");
+    true
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
Would it be good to already put this in a separate file? Seems to me we're going to do that anyways the moment we start jamming.

Reply:
I was using [the normal Rust idiom](https://doc.rust-lang.org/rust-by-example/testing/unit_testing.html) of having unit tests at the bottom of the same file as the functions they are testing.

I believe the main reason this is done is to be able to access to private functions as well.

Reply:
Cool, makes sense I suppose, thanks for explaining it to me :)